sdk/storage/azblob/blockblob/client_test.go

```diff
@@ -3975,3 +3977,40 @@ func TestUploadBufferEvenBlockSize(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, int64(len(buffer)), fbb.totalStaged)
 }
+
+func TestUploadLogEvent(t *testing.T) {
+	listnercalled := false
+	log.SetEvents(azblob.EventUpload, log.EventRequest, log.EventResponse)
+	log.SetListener(func(cls log.Event, msg string) {
+		t.Logf("%s: %s\n", cls, msg)
+		listnercalled = true
+	})
+
+	fbb := &fakeBlockBlob{}
+	client, err := blockblob.NewClientWithNoCredential("https://fake/blob/path", &blockblob.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			Transport: fbb,
+			Telemetry: policy.TelemetryOptions{ApplicationID: "testApp/1.0.0-preview.2"},
+		},
+	})
+	require.NoError(t, err)
+	require.NotNil(t, client)
+
+	// create fake source that's evenly divisible by 50000 (max number of blocks)
+	// and greater than MaxUploadBlobBytes (256MB) so that it doesn't fit into a single upload.
+
+	buffer := make([]byte, 270000000)
+	for i := 0; i < len(buffer); i++ {
+		buffer[i] = 1
+	}
+
+	_, err = client.UploadBuffer(context.Background(), buffer, &blockblob.UploadBufferOptions{
+		Concurrency: 1,
+		Progress: func(bytesTransferred int64) {
+			t.Logf("%v percent job done", (bytesTransferred*100)/270000000)
+		},
+	})
+	require.NoError(t, err)
+	require.Equal(t, int64(len(buffer)), fbb.totalStaged)
+	require.Equal(t, listnercalled, true)
```

Comment:
I would tweak the test to validate that the logged string matches your expectations.  This also helps reviewers see the exact format of the message.

Reply:
check added as per comment